### PR TITLE
Fix dependencies and add verify-deps

### PR DESCRIPTION
**Pull request by tnozicka**
/cc @mfojtik  @soltysh @sanchezl

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+	k8s.io/client-go v0.17.0
+	k8s.io/component-base v0.17.0
+	k8s.io/klog v1.0.0
+)
+
+replace (
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20191217141120-791af96035a5
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20191216194936-57f413491e9e
+	github.com/openshift/library-go => github.com/openshift/library-go v0.0.0-20191218095328-1c12909e5923
+	k8s.io/api => k8s.io/api v0.17.0
```
Comment (deads2k) [inline go.mod]: why is replace needed?  it shouldn't be needed.

File: go.mod [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
-	k8s.io/kube-aggregator v0.0.0-20190918161219-8c8f079fddc3 // indirect
+	k8s.io/api v0.17.0
+	k8s.io/apimachinery v0.17.0
+	k8s.io/client-go v0.17.0
+	k8s.io/component-base v0.17.0
+	k8s.io/klog v1.0.0
+)
+
+replace (
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20191217141120-791af96035a5
```
Comment (deads2k) [inline go.mod]: I didn't expect to need a replace

File: go.mod (hunk 1 above)
Comment (tnozicka) [reply go.mod]: I thought we are replacing so it gets pinned to correct branch even when someone transitively goes higher, that patter is established in our other repos but I am open to how we prefer to handle it.

Comment (sanchezl) [reply go.mod]: We've been using it to pin non-semver, non-master dependencies. So, it shouldn't be needed in this case.

**Pull request merged**